### PR TITLE
refactor: A32NX BatDisplay to fbw-common / add A380 separate BAT instrument

### PR DESCRIPTION
**Pull request by frankkopp**
## Summary of Changes

- [x] Moved the A32NX BatDisplay code to fbw-common.
- [x] Added a separate instrument for the A380X.
- [x] Added a LVAR for the display knob (animation not modeled in cockpit yet) to display one of the 4 batteries or being turned OFF. 
- [X] Added B: events 

L:A380X_KNOB_OVHD_BAT_POSITION: 0..4

## Screenshots (if necessary)
https://github.com/flybywiresim/aircraft/assets/16833201/96a17cfd-8921-4b6e-a4aa-c2261dcb0354

Discord username (if different from GitHub): cdr_maverick

## Testing instructions
A380 only
Test if the BAT Voltage Indicator Switch and display work as expected. 
It should show:
L:A32NX_ELEC_BAT_${batteryNumber}_POTENTIAL
according to this mapping:
batteryNumber = ESS=4,  APU=3, OFF=00.0V, BAT1=1, BAT2=2

If you you know how you can check the assosiated LVar and B: Events
- L:A380X_OVHD_ELEC_BAT_SELECTOR_KNOB

image

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by frankkopp** [comment]
> When turning it it should read 0.00V: A380 Voltmeter showing 0.00V
> 
> Something about the size/aspect ratio doesn't look right though, it seems like there is uniform padding around the characters whereas in our version there's a lot of padding left and right.
> 
> Also just to confirm, it seems the char-by-char update from the 320 still exists, so that can stay: [youtu.be/GspxOmtreIQ?t=43](https://youtu.be/GspxOmtreIQ?t=43)

Fixed the OFF = 00.0V

CSS I will work on - trying to find a better reference

**Comment by tracernz** [comment]
> Something about the size/aspect ratio doesn't look right though, it seems like there is uniform padding around the characters whereas in our version there's a lot of padding left and right.

The display in the A320 model is the wrong aspect ratio and size, so the instrument is setup to match and incorrect screen and look as good as possible given that issue.

**Comment by beheh** [comment]
> The display in the A320 model is the wrong aspect ratio and size, so the instrument is setup to match the incorrect screen and look as good as possible given that issue.

It's much better already in the video at the top of the PR right now! It originally looked like this: 
image

**Comment by frankkopp** [comment]
> this:
> 

Yea - tweaked it a bit alredy. 

Do we have real life reference - I haven't found much. 
Also is this display always on independent of any power bus?

File: fbw-a380x/src/systems/instruments/src/BAT/style.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  font-style: normal;
+}
+
+.bat-svg {
+  background-color: hsl(55, 82%, 2%);
+  box-shadow: inset 0px 0px 110px -20px rgba(255, 175, 0, 0.25);
+  height: 100%;
+  width: 100%;
+  left: 0;
+  /* TODO: Remove opacity when SAI/ISIS/CLOCK/BAT/ATC/RMP/RUD TRIM is split and not share the same emissive/mesh */
```
Comment (tracernz) [inline fbw-a380x/src/systems/instruments/src/BAT/style.scss:16]: This comment from the A32NX is probably not relevant for the A380.

**Review by tracernz — APPROVED** [review]

**Pull request merged**